tests/contrib/httpx/test_httpx.py

```diff
@@ -0,0 +1,315 @@
+import os
+
+import httpx
+import pytest
+
+from ddtrace import config
+from ddtrace.contrib.httpx.patch import patch
+from ddtrace.contrib.httpx.patch import unpatch
+from ddtrace.settings.http import HttpConfig
+from ddtrace.vendor.wrapt import ObjectProxy
+from tests.utils import override_config
+from tests.utils import override_http_config
+
+
+# host:port of httpbin_local container
+HOST = "localhost"
+PORT = 8001
+
+
+def get_url(path):
+    # type: (str) -> str
+    return "http://{}:{}{}".format(HOST, PORT, path)
+
+
+@pytest.fixture(autouse=True)
+def patch_httpx():
+    patch()
+    try:
+        yield
+    finally:
+        unpatch()
+
+
+def test_patching():
+    """
+    When patching httpx library
+        We wrap the correct methods
+    When unpatching httpx library
+        We unwrap the correct methods
+    """
+    assert isinstance(httpx.Client.send, ObjectProxy)
```

Comment:
Why do we wrap send methods in an ObjectProxy? Is this our way of creating a tracer interceptor?

Reply:
that is correct, we use the [wrapt](https://wrapt.readthedocs.io/en/latest/) library to create wrappers around methods/functions/classes and this is how we do our instrumentation.

the base class used by `wrapt` is `ObjectProxy`.

this is where we create the function wrappers and override the default methods: https://github.com/DataDog/dd-trace-py/blob/896edc1f5281e807e916792ffb7d281aa861f07f/ddtrace/contrib/httpx/patch.py#L138-L139

`_w` is a `wrapt.wrap_function_wrapper`